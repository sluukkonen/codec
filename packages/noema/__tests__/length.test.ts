import { array, length, number } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

const smallNumberArray = length(array(number), 1, 2)

test("delegates to the underlying codec if the length is between the minimum and maximum length", () => {
  expectParseSuccess(smallNumberArray, [1])
  expectParseSuccess(smallNumberArray, [1, 2])
})

test("delegates to the underlying codec if length is missing or is not an integer", () => {
  expectParseFailure(smallNumberArray, null)
  expectParseFailure(smallNumberArray, { length: "0" })
  expectParseFailure(smallNumberArray, { length: NaN })
})

test("fails to parse if the length is below the minimum length", () => {
  expectParseFailure(smallNumberArray, [])
})

test("fails to parse if the length is above the maximum length", () => {
  expectParseFailure(smallNumberArray, [1, 2, 3])
})
