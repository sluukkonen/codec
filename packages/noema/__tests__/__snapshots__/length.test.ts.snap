// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`length delegates to the underlying codec if length is missing or is not an integer 1`] = `
[
  {
    "code": "invalid_array",
    "path": [],
  },
]
`;

exports[`length delegates to the underlying codec if length is missing or is not an integer 2`] = `
[
  {
    "code": "invalid_array",
    "path": [],
  },
]
`;

exports[`length delegates to the underlying codec if length is missing or is not an integer 3`] = `
[
  {
    "code": "invalid_array",
    "path": [],
  },
]
`;

exports[`length fails to parse if the length is above the maximum length 1`] = `
[
  {
    "code": "too_long",
    "length": 3,
    "maxLength": 2,
    "path": [],
  },
]
`;

exports[`length fails to parse if the length is below the minimum length 1`] = `
[
  {
    "code": "too_short",
    "length": 0,
    "minLength": 1,
    "path": [],
  },
]
`;
